wrappers/rust/oneds-telemetry/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+use log::debug;
+use once_cell::sync::Lazy;
+use std::{sync::RwLock, ffi::CString};
+use types::{StringProperty, TelemetryItem, TelemetryProperty};
+use cpp_client_telemetry_sys::{
+    evt_open,
+    evt_close,
+    evt_log,
+    evt_flush,
+    evt_upload,
+    evt_prop
+};
+
+pub mod appender;
+pub mod types;
+
+#[derive(Clone)]
+pub struct LogManager {
+    is_started: bool,
+    configuration: CString,
+    log_handle: Option<i64>,
+}
+
+/// The global `Tracer` provider singleton.
+static GLOBAL_LOG_MANAGER: Lazy<RwLock<LogManager>> = Lazy::new(|| RwLock::new(LogManager::new()));
```

Comment:
Is there a reason this uses `Lazy` instead of just typing this field as `RwLock<LogManager>`?